core/camel-support/src/main/java/org/apache/camel/support/component/EndpointUriFactorySupport.java

```diff
@@ -78,6 +79,20 @@ protected String buildQueryParameters(String uri, Map<String, Object> parameters
             throws URISyntaxException {
         // we want sorted parameters
         Map<String, Object> map = new TreeMap<>(parameters);
+        map.replaceAll(new BiFunction<String, Object, Object>() {
```

Comment:
Isnt there a way to do this faster, eg now you replace all values? What if there are no secrets at all? Wont it be faster just to walk on the secret property names, and see if they are in the parameters, and then only replace those.